### PR TITLE
some clippy and warnings and more checks

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
- added deny of multi version as warning (cleaned up serde) - so we avoid compile several deps many times
- made clippy hard error in lending
- some small clean ups

Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -160,9 +161,17 @@ jobs:
...
       - name: Cargo clippy
         run: |
-          SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --all-targets 
+          SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --all-targets
+
+      - name: Cargo Single Depedency
+        description: Speed up builds and reduced security check surface
+        run: |
+         /home/runner/.cargo/bin/cargo install -f cargo-deny
+         /home/runner/.cargo/bin/cargo deny check  --hide-inclusion-graph Ban
```
Comment (seunlanlege) [inline .github/workflows/check.yml:173]:
```suggestion
         cargo install -f cargo-deny
         cargo deny check  --hide-inclusion-graph Ban
```

**Comment by seunlanlege** [comment]
@dzmitry-lahoda you've broken our check.yml file thats why there are no other checks running

**Comment by dzmitry-lahoda** [comment]
may be there is no need to run simnode on review prs?

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -160,9 +161,17 @@ jobs:
...
       - name: Cargo clippy
         run: |
-          SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --all-targets 
+          SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --all-targets
+
+      - name: Cargo Single Depedency
+        # description: Speed up builds and reduced security check surface
+        run: |
+         cargo install -f cargo-deny
+         cargo deny check  --hide-inclusion-graph Ban
```
Comment (andor0) [inline .github/workflows/check.yml:173]: 2 spaces between `check` and  `--hide-inclusion-graph`

**Review by seunlanlege — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Pull request merged**